docs: consistently name Consul service mesh (#17222)
hashicorp/consul: website/content/docs/connect/proxies/envoy.mdx

# Envoy Proxy Configuration for Service Mesh

> Consul servic emesh has first class support for using

Reviewer: ```suggestion
Consul service mesh has first class support for using
```

- Configuration to [expose specific HTTP paths](/consul/docs/connect/registration/service-registration#expose-paths-configuration-reference).

For more information on the parts of the Envoy proxy runtime configuration
that are currently controllable via Consul service mesh, refer to [Dynamic
Configuration](#dynamic-configuration).

We plan to enable more and more of Envoy's features through

> Consul's first-class configuration over time, however some advanced users will

Reviewer: ```suggestion
Consul service mesh's first-class configuration over time, however some advanced users will
```

Reviewer: not sure if we need to be explicit here, feel free to ignore

entry](/consul/docs/connect/config-entries/proxy-defaults) to act as
defaults that are inherited by all services.

> - `protocol` - The protocol the service speaks. Consul's Envoy integration

Reviewer: ```suggestion
- `protocol` - The protocol the service speaks. Consul service mesh's Envoy integration
```

- `envoy_extra_static_listeners_json` - Similar to
  `envoy_extra_static_clusters_json` but appends one or more [Envoy listeners](https://www.envoyproxy.io/docs/envoy/v1.17.2/api-v3/config/listener/v3/listener.proto) to the array of [static
  listener](https://www.envoyproxy.io/docs/envoy/v1.17.2/api-v3/config/bootstrap/v3/bootstrap.proto#envoy-v3-api-field-config-bootstrap-v3-bootstrap-staticresources-listeners) definitions.

>   Can be used to setup limited access that bypasses service mesh mTLS or

Reviewer: ```suggestion
  Can be used to setup limited access that bypasses the service mesh's mTLS or
```
